lib/charms/traefik_k8s/v0/ingress_per_unit.py

```diff
@@ -186,13 +431,102 @@ def is_failed(self, relation: Relation = None):
                 raise RelationDataMismatchError(relation, unit)
         return False
 
+    def get_request(self, relation: Relation):
+        """Get the IngressRequest for the given Relation."""
+        return IngressRequest(self, relation, self._fetch_ingress_data(relation))
+
+    def _fetch_ingress_data(self, relation: Relation, validate=False):
+        """Fetch and validate the databags.
+
+        For the provider side: the application databag.
+        For the requirer side: the unit databag.
+        """
+        this_unit = self.unit
+        this_app = self.app
+
+        if not relation.app or not relation.app.name:
+            # Handle edge case where remote app name can be missing.
+            return {relation.app: {}, this_app: {}, this_unit: {}}
+
+        ingress_data: dict = {}
+        # we start by looking at the provider's app databag
+        if this_unit.is_leader():
+            # only leaders can read their app's data
+            data = relation.data[this_app].get("data")
+            deserialized = {}
+            if data:
+                deserialized = _deserialize_data(data)
+                if validate:
+                    _validate_data(deserialized, INGRESS_PROVIDES_APP_SCHEMA)
```

Comment:
Who is responsible for catching the exception? If not here then should probably add a `Raises:` statement to the public `get_request` docstring.